index.html

```diff
@@ -43,6 +43,15 @@
     <meta name="msapplication-TileImage" content="<%= require('./images/manifest/icon-144x144.png') %>">
     <meta name="msapplication-TileColor" content="#4b4b4b">
     <meta name="msapplication-tap-highlight" content="no">
+
+    <!-- Load and register service worker that adds offline support -->
+    <script>
+      if ('serviceWorker' in navigator) {
+        window.addEventListener('load', function() {
+          navigator.serviceWorker.register('/service-worker.js');
```

Comment:
Since I'm moving toward hosting on a CDN, where long-term caching means we need to do some kind of cache-busting, what happens if we want to make an update to this file? Do we manually rename it? Is it important that it have this exact name?

If it's okay for it to have an arbitrary name, and to change names any time it gets updated, I can look into using the same `require` syntax we use elsewhere. It will be a bit more complex since the file is generated rather than being an existing file, but I think I can probably figure it out.

Reply:
I answered this question for myself below.